platforms/ide/ide/src/main/java/org/gradle/plugins/ide/eclipse/model/EclipseProject.java

```diff
@@ -135,23 +140,33 @@ public abstract class EclipseProject {
     public static final ImmutableSet<String> VALID_LINKED_RESOURCE_ARGS = ImmutableSet.of("name", "type", "location", "locationUri");
     private String name;
 
-    private String comment;
+    private Property<String> comment;
 
     private Set<String> referencedProjects = new LinkedHashSet<>();
 
     private List<String> natures = new ArrayList<>();
 
     private List<BuildCommand> buildCommands = new ArrayList<>();
 
-    private Set<Link> linkedResources = new LinkedHashSet<>();
+    private final SetProperty<Link> linkedResources;
 
     private Set<ResourceFilter> resourceFilters = new LinkedHashSet<>();
 
     private final XmlFileContentMerger file;
 
     @Inject
-    public EclipseProject(XmlFileContentMerger file) {
+    public EclipseProject(XmlFileContentMerger file, org.gradle.api.Project project) {
```

Comment:
❓ we probably can avoid changing the constructor if we inject the project reference on a method

```
@Inject
Project getProject() { 
    throw new UnsupportedOperationException();
}
``